sdk/ml/azure-ai-ml/azure/ai/ml/operations/_feature_store_operations.py

```diff
@@ -317,13 +322,15 @@ def begin_update(
                 not existing_offline_store_connection.properties
                 or existing_offline_store_connection.properties.target != offline_store.target
             ):
+                update_offline_store_connection = True
                 update_offline_store_role_assignment = True
                 module_logger.info(
                     "Warning: You have changed the offline store connection, "
                     "any data that was materialized "
                     "earlier will not be available. You have to run backfill again."
                 )
         elif offline_store_target_to_update:
+            update_offline_store_connection = True
```

Comment:
Why do we set it to True here?

Reply:
According to this logic, we will create a new connection even if `offline_store_target_to_upadate == offline_store_target_to_update`

Reply:
No. the logic here is: when the existing feature store does not have the connection, we will create one connection for it.